plasmapy/formulary/parameters.py

```diff
@@ -663,23 +689,27 @@ def thermal_speed(
     >>> thermal_speed(1e6*u.K, "e-", method="mean_magnitude")
     <Quantity 621251... m / s>
     """
-    m = mass if np.isfinite(mass) else particles.particle_mass(particle)
+    if mass is None:
+        mass = particles.particle_mass(particle)
 
     # different methods, as per https://en.wikipedia.org/wiki/Thermal_velocity
     try:
-        coef = _coefficients[ndim]
+        coeff = thermal_speed_coefficients[ndim]
     except KeyError:
         raise ValueError("{ndim} is not a supported value for ndim in thermal_speed")
     try:
-        coef = coef[method]
+        coeff = coeff[method]
```

Comment:
continues minor refactor
```suggestion
        coeff = coeffs_for_dim[method]
```